app/crud/wh_sched.py

```diff
@@ -277,6 +294,42 @@ def convert_time_str(time_str) -> datetime.time:
     return datetime.datetime.strptime(time_str, "%I:%M %p").time()
 
 
+def fetch_schedules_with_defaults(
+    session: Session, warehouse: str
+) -> List[WarehouseSchedules]:
+    """
+    Returns the WarehouseSchedules for the given warehouse.
+    :param session: Snowpark session instance.
+    :param warehouse: The snowflake warehouse to filter on.
+    :return: A list of WarehouseSchedules for the given warehouse, creating and persisting
+    default schedules if none exist.
+    """
+    schedules = WarehouseSchedules.find_all(session, warehouse)
+    if len(schedules) == 0:
+        wh = describe_warehouse(session, warehouse)
+        wh.write(session)
+        schedules.append(wh)
+        wh2 = describe_warehouse(session, warehouse)
+        wh2.weekday = False
+        wh2.write(session)
+        schedules.append(wh2)
+    elif len(schedules) == 2 and all(not s.enabled for s in schedules):
```

Comment:
I'm going to have to come up with a better way to express this.

I didn't want to change the `internal.wh_schedules` table schema but I feel like that might be the correct thing to do when a user has edited the schedule. As this is, you can't create a schedule which sets the warehouse size to something other than the default (as it will be overwritten on update).

Reply:
fixed in latest patch.